Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM ubuntu:focal
```

Comment:
I think basic idea for CI/CD should be that you should have a build stage which builds (only if the Dockerfile is changed). and then the next stage builds would use the previous build and build for source and use that to run tests.